src/main/resources/templates/db_scripts/schema.json

```diff
@@ -488,6 +488,12 @@
           "tOps": "ADD",
           "caseSensitive": false,
           "removeAccents": true
+        },
+        {
+          "fieldName": "natureOfContentTermIds",
```

Comment:
Why is an index being added?

@skoczko Is it now standard practice to add indexes for new properties?

Are we intending for this property to be used in filtering, searching or sorting?

Reply:
I've decided to add this index because potentially we may require to search through those fields in the future. Should I remove it for now?

Reply:
@bohdan-suprun  

> I've decided to add this index because potentially we may require to search through those fields in the future.

My concern with this approach is that any property is potentially searchable with the way FOLIO uses CQL which means that we could end up creating indexes for every property in a record (with likely significant write performance impact).

That said, deferring index creation has often caused performance surprises later on.

@skoczko @adamdickmeiss @julianladisch What is the guidance around this, should we pre-emptively create indexes?

If so, what kind of index is best suited to an array of UUID strings?

Reply:
@bohdan-suprun @marcjohnson-kint Unless it is known that the property will be used for searching, filtering or sorting (e.g there is a feature/user story somewhere in the backlog that will call for this functionality) I would recommend against pre-emptively adding indexes. We now have a "warning" system in RMB which will produce a WARN in the log if a property is used for searching but there is no index.

For UUIDs I would recommend btree (regular "index"), it's the smallest and fastest index type and also allows sorting.

Reply:
@skoczko Thanks. @bohdan-suprun I think that is your answer, we should not create an index at this point.

I think that also makes @mreno-EBSCO point about database migrations redundant for this particular change. It is still a very important topic to consider.

Reply:
Hi @skoczko, @marcjohnson-kint, @mreno-EBSCO, I've removed the index from schema.json file, so resolving this discussion.